File: gluecodium/src/test/resources/smoke/comments/output/android/com/example/smoke/ExcludedCommentsOnly.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
/**
 * @hidden
 */
public final class ExcludedCommentsOnly extends NativeBase {
    /**
     * @hidden
     */
    public static final boolean VERY_USEFUL = true;
    /**
     * @hidden
     */
    public enum SomeEnum {
        /**
         * @hidden
         */
        USELESS(0);
        public final int value;
        SomeEnum(final int value) {
            this.value = value;
        }
    }
    /**
     * @hidden
     */
    public static final class SomethingWrongException extends Exception {
        public SomethingWrongException(final ExcludedCommentsOnly.SomeEnum error) {
            super(error.toString());
            this.error = error;
        }
        public final ExcludedCommentsOnly.SomeEnum error;
    }
    /**
     * @hidden
     */
    public static final class SomeStruct {
        /**
         * @hidden
         */
        public boolean someField;
        public SomeStruct(final boolean someField) {
            this.someField = someField;
        }
    }
    /**
     * @hidden
     */
    static class SomeLambdaImpl extends NativeBase implements SomeLambda {
        protected SomeLambdaImpl(final long nativeHandle, final Object dummy) {
            super(nativeHandle, new Disposer() {
                @Override
                public void disposeNative(long handle) {
                    disposeNativeHandle(handle);
                }
            });
        }
        private static native void disposeNativeHandle(long nativeHandle);
        /**
         *
         * @hidden
         * @param p0
         * @param index
         * @return
         */
        public native double doIt(@NonNull final String p0, final int index);
    }
    /**
     * @hidden
     */
    @FunctionalInterface
    public interface SomeLambda {
        /**
         *
         * @hidden
         * @param p0
         * @param index
         * @return
         */
        double doIt(@NonNull final String p0, final int index);
    }
    /**
     * For internal use only.
     * @hidden
     */
    protected ExcludedCommentsOnly(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    /**
     *
     * @hidden
     * @param inputParameter
     * @return
     * @throws ExcludedCommentsOnly.SomethingWrongException
     */
    public native boolean someMethodWithAllComments(@NonNull final String inputParameter) throws ExcludedCommentsOnly.SomethingWrongException;
    /**
     *
     * @hidden
     */
    public native void someMethodWithoutReturnTypeOrInputParameters();
    /**
     *
     * @hidden
     */
    public native boolean isSomeProperty();
    /**
     *
     * @hidden
     * @param value
     */
    public native void setSomeProperty(final boolean value);
}

File: gluecodium/src/test/resources/smoke/comments/output/android-kotlin/com/example/smoke/ExcludedCommentsOnly.kt
/*

 *
 */

@file:JvmName("ExcludedCommentsOnlyExtensions")


package com.example.smoke

import com.example.NativeBase

class ExcludedCommentsOnly : NativeBase {

    enum class SomeEnum(@JvmField val value: Int) {
        USELESS(0);
    }
    class SomethingWrongException(@JvmField val error: ExcludedCommentsOnly.SomeEnum) : Exception(error.toString())


    class SomeStruct {
        @JvmField var someField: Boolean



        constructor(someField: Boolean) {
            this.someField = someField
        }





    }

    fun interface SomeLambda {
        /**
         *
         * @suppress
         * @param p0
         * @param p1
         * @return
         */

        fun apply(p0: String, p1: Int) : Double
    }

    /**
     * @suppress
     *
     * This class is used to represent C++ implementations of the interface or lambda in Kotlin.
     * It is instantiated by JNI and should not be used by the end users.
     */
    private class SomeLambdaImpl : NativeBase, SomeLambda {
        protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
            : super(nativeHandle, { disposeNativeHandle(it) }) {}

        /**
         *
         * @suppress
         * @param p0
         * @param p1
         * @return
         */

        override external fun apply(p0: String, p1: Int) : Double



        companion object {
            @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
        }
    }


    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}



    /**
     *
     * @suppress
     * @param inputParameter
     * @return
     * @throws ExcludedCommentsOnly.SomethingWrongException
     */
    @Throws(ExcludedCommentsOnly.SomethingWrongException::class)
    external fun someMethodWithAllComments(inputParameter: String) : Boolean
    /**
     *
     * @suppress
     */

    external fun someMethodWithoutReturnTypeOrInputParameters() : Unit

    var isSomeProperty: Boolean
        external get
        external set




    companion object {
        @JvmField final val VERY_USEFUL: Boolean = true
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}


